pkg/cloudprovider/aws/amifamily/bootstrap/eksbootstrap.go

```diff
@@ -73,12 +94,91 @@ func (e EKS) nodeLabelArg() string {
 	nodeLabelArg := ""
 	labelStrings := []string{}
 	var once sync.Once
-	for k, v := range e.Labels {
-		if v1alpha5.LabelDomainExceptions.Has(k) {
+	keys := make([]string, 0, len(e.Labels))
+	for k := range e.Labels {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys) // ensures this list is deterministic, for easy testing.
+	for _, key := range keys {
+		if v1alpha5.LabelDomainExceptions.Has(key) {
```

Comment:
How could these sneak in here? Shouldn't we filter them from our the vendor neutral code?

Reply:
This is probably related to what Felix found. We let you use the `kops.k8s.io` [on the node object](https://github.com/aws/karpenter/blob/main/pkg/apis/provisioning/v1alpha5/labels.go#L88), but filter it out here, otherwise the kubelet would fail on start up.

Reply:
Let's catch this at the root and treat it as an invariant to avoid too much knowledge here.

Reply:
I'd rather not do it in this PR and keep the changes focused on just AL2 merging. https://github.com/aws/karpenter/issues/1883 seems like the right place for this.

Reply:
@felix-zhe-huang's change is already merged, so you should be able to leave this out and trust the invariant.